edi/models/edi_exchange_consumer_mixin.py

```diff
@@ -19,6 +25,104 @@ class EDIExchangeConsumerMixin(models.AbstractModel):
         domain=lambda r: [("model", "=", r._name)],
     )
     exchange_record_count = fields.Integer(compute="_compute_exchange_record_count")
+    expected_edi_configuration = fields.Serialized(
+        compute="_compute_expected_edi_configuration"
+    )
+    has_expected_edi_configuration = fields.Boolean(
+        compute="_compute_expected_edi_configuration"
+    )
+
+    def _compute_expected_edi_configuration(self):
+        for record in self:
+            configurations = record._get_expected_edi_configuration()
+            record.expected_edi_configuration = configurations
+            record.has_expected_edi_configuration = bool(configurations)
+
+    def _get_expected_edi_configuration(self):
+        exchange_types = (
+            self.env["edi.exchange.type"]
+            .sudo()
+            .search([("model_ids.model", "=", self._name), ("backend_id", "!=", False)])
+        )
+        result = {}
+        for exchange_type in exchange_types:
+            if exchange_type.enable_domain and not self.search(
+                [("id", "=", self.id)]
+                + safe_eval(exchange_type.enable_domain, self._get_eval_context())
+            ):
+                continue
+            if exchange_type.enable_snippet:
+                eval_ctx = {
+                    "record": self,
+                    "exchange_type": exchange_type,
+                }
+                eval_ctx.update(self._get_eval_context())
+                safe_eval(
+                    exchange_type.enable_snippet, eval_ctx, mode="exec", nocopy=True
+                )
+                if not eval_ctx.get("result", False):
+                    continue
+            result[exchange_type.id] = exchange_type.display_name
```

Comment:
do not search for the record, you have it.  Use `filtered_domain` to apply the domain on it:
```suggestion
        for exchange_type in exchange_types:
            eval_ctx = dict(
                self._get_eval_context(),
                record=self,
                exchange_type=exchange_type
            )
            domain = safe_eval(
                exchange_type.enable_domain or "[]", eval_ctx
            )
            if not self.filtered_domain(domain):
                continue
            if exchange_type.enable_snippet:
                safe_eval(
                    exchange_type.enable_snippet, eval_ctx, mode="exec", nocopy=True
                )
                if not eval_ctx.get("result", False):
                    continue
            result[exchange_type.id] = exchange_type.display_name
```